packages/vite/src/lib/entries.ts

```diff
@@ -0,0 +1,49 @@
+import path from 'node:path'
+
+import type { PagesDependency } from '@redwoodjs/project-config'
+import {
+  ensurePosixPath,
+  getPaths,
+  processPagesDir,
+} from '@redwoodjs/project-config'
+
+const getPathRelativeToSrc = (maybeAbsolutePath: string) => {
+  // If the path is already relative
+  if (!path.isAbsolute(maybeAbsolutePath)) {
+    return maybeAbsolutePath
+  }
+
+  return `./${path.relative(getPaths().web.src, maybeAbsolutePath)}`
+}
+
+const withRelativeImports = (page: PagesDependency) => {
+  return {
+    ...page,
+    relativeImport: ensurePosixPath(getPathRelativeToSrc(page.importPath)),
+  }
+}
+
+export function getEntries() {
+  const entries: Record<string, string> = {}
+
+  // Add the various pages
+  const pages = processPagesDir().map(withRelativeImports)
+  for (const page of pages) {
+    entries[page.importName] = page.path
+  }
+
+  // Add the "ServerEntry" entry
+  const serverEntry = getPaths().web.entryServer
+  if (!serverEntry) {
+    throw new Error('Server Entry file not found')
+  }
+  entries['ServerEntry'] = serverEntry
```

Comment:
Thoughts on using some prefix here like `_rw_ServerEntry` and then using that reference when we look it up in `clientSsr` and such?

Reply:
To minimize risk of clashing names? Sure! Unfortunately we're not super consistent `_rw_something`, `__rw__something`, `_rwjs_something`, `__rwjs__something` etc...
I'd want us to move to double underscore at the start at least.
For constants and env vars I also want it after the prefix because there we use single underscore for the "something" part if it's more than one word. Like we have `__RWJS__TEST_IMPORTS` which I like. There we also use rwjs. 
For lower case stuff in the code we do more of `__rw_something` currently. But maybe we should move towards `__rwjs__something` to match what we do for env vars.

What do you think?

Reply:
I would like to use double underscore before and after the redwood identifier. I don't really have an opinion on `rw` vs `rwjs` so happy for you to just pick one - sounds like `__rwjs__xxx` is the preference then?

Reply:
Addressed in: [8fa8b26](https://github.com/redwoodjs/redwood/pull/10533/commits/8fa8b26f5445dc9353473f814707bd39c48e4700)